### PR TITLE
add VideoPlayer.offset() infolabels

### PR DESCRIPTION
**Pull request by ronie**
this adds VideoPlayer.offset() and VideoPlayer.position() support for VideoPlayer infolabels.
it allows skins to display info for (amongst others) the next item during playback.

offset() is relative to the currently playing video in a playlist
position() is the absolute position in a playlist

@sualfred @gade01 @jjd-uk

### CODE REVIEW

**Comment by jjd-uk** [comment]
Looks to be working as expected

image

Thanks ronie

File: xbmc/GUIInfoManager.cpp [hunk 1]
```diff
@@ -9488,6 +9967,16 @@ int CGUIInfoManager::TranslateMusicPlayerString(const std::string &info) const
   return 0;
 }
 
+int CGUIInfoManager::TranslateVideoPlayerString(const std::string &info) const
```
Comment (ksooo) [inline xbmc/GUIInfoManager.cpp]: coding guidelines: left align `&`

File: xbmc/GUIInfoManager.cpp [hunk 2]
```diff
@@ -9382,6 +9845,22 @@ int CGUIInfoManager::TranslateSingleString(const std::string &strCondition, bool
         return AddMultiInfo(CGUIInfo(value, 1, position)); // 1 => relative
       }
     }
+
```
Comment (ksooo) [inline xbmc/GUIInfoManager.cpp]: please remove this empty line

File: xbmc/GUIInfoManager.h [hunk 3]
```diff
@@ -173,6 +173,7 @@ class CGUIInfoManager : public KODI::MESSAGING::IMessageTarget
   int TranslateSingleString(const std::string &strCondition);
   int TranslateListItem(const Property& cat, const Property& prop, int id, bool container);
   int TranslateMusicPlayerString(const std::string &info) const;
+  int TranslateVideoPlayerString(const std::string &info) const;
```
Comment (ksooo) [inline xbmc/GUIInfoManager.h]: coding guidelines: left align `&`

File: xbmc/guilib/guiinfo/VideoGUIInfo.cpp [hunk 4]
```diff
@@ -579,6 +586,48 @@ bool CVideoGUIInfo::GetLabel(std::string& value, const CFileItem *item, int cont
   return false;
 }
 
+bool CVideoGUIInfo::GetPlaylistInfo(std::string& value, const CGUIInfo &info) const
```
Comment (ksooo) [inline xbmc/guilib/guiinfo/VideoGUIInfo.cpp]: coding guidelines: left align `&`

File: xbmc/guilib/guiinfo/VideoGUIInfo.h [hunk 5]
```diff
@@ -35,6 +35,7 @@ class CVideoGUIInfo : public CGUIInfoProvider
 
 private:
   int GetPercentPlayed(const CVideoInfoTag* tag) const;
+  bool GetPlaylistInfo(std::string& value, const CGUIInfo &info) const;
```
Comment (ksooo) [inline xbmc/guilib/guiinfo/VideoGUIInfo.h]: coding guidelines: left align `&`

**Comment by gade01** [comment]
Tested all kinds of scenarios. And everything worked as expected.
Thanks!

**Comment by ronie** [comment]
thanx @ksooo, updated as suggested.

**Pull request merged**